diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -89,42 +95,82 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     private lateinit var positionByOffset: (Int) -> Pair<Int, Int>
 
     override fun logic(node: ASTNode) {
-        if (node.elementType == KtFileElementType.INSTANCE) {
-            node.getChildren(null).forEach {
-                if (it.elementType != PACKAGE_DIRECTIVE && it.elementType != IMPORT_LIST) {
-                    checkLength(it, configuration)
+        var isFixedSmthInPreviousStep: Boolean
+
+        do {
+            isFixedSmthInPreviousStep = false
+
+            if (node.elementType == KtFileElementType.INSTANCE) {
+                node.getChildren(null).forEach {
+                    if (it.elementType != PACKAGE_DIRECTIVE && it.elementType != IMPORT_LIST) {
+                        val isFixedSmthInChildNode = checkLength(it, configuration)
+
+                        if (!isFixedSmthInPreviousStep && isFixedSmthInChildNode) {
+                            isFixedSmthInPreviousStep = true
```

Comment:
No, this can't be changed, because this code inside ` forEach` and checks whether at least one of node children has been fixed at any of  `forEach` step.